src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -79,15 +83,62 @@ public static List<WifiScanEntry> parseWifiScanEntries(
 			for (JsonElement e : scanInfo) {
 				WifiScanEntry entry = gson.fromJson(e, WifiScanEntry.class);
 				entry.unixTimeMs = timestampMs;
+				extractIEs(e, entry);
 				entries.add(entry);
-
 			}
 		} catch (Exception e) {
+			logger.debug("Exception when parsing wifiscan entries", e);
 			return null;
 		}
 		return entries;
 	}
 
+	/**
+	 * Extract desired information elements (IEs) from the wifiscan entry.
+	 * Modifies {@code entry} argument. Skips invalid IEs (IEs with missing
+	 * fields).
+	 */
+	private static void extractIEs(
+		JsonElement entryJsonElement,
+		WifiScanEntry entry
+	) {
+		JsonElement iesJsonElement =
+			entryJsonElement.getAsJsonObject().get("ies");
+		if (iesJsonElement == null) {
+			logger.debug("Wifiscan entry does not contain 'ies' field.");
+			return;
+		}
+		JsonArray ies = iesJsonElement.getAsJsonArray();
+		for (JsonElement ie : ies) {
+			JsonElement typeElement = ie.getAsJsonObject().get("type");
+			if (typeElement == null) {
+				continue;
+			}
+			final String contentFieldKey = "content";
```

Comment:
I'd prefer to make it a static final for the class, but feel free to push me back.

Reply:
I'll make it `private static final`